src/App.tsx

```diff
@@ -1,9 +1,38 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
+export interface Movie {
+  title: string;
+  description: string;
+  imgUrl: string;
+  imdbUrl: string;
+  imdbId: string;
+}
+
+export function filterMovies(movies: Movie[], inputValue: string) {
+  if (inputValue === '') {
+    return movies;
+  }
+
+  return movies.filter(movie => {
+    return (
+      movie.title
+        .toLocaleLowerCase()
+        .includes(inputValue.trim().toLocaleLowerCase()) ||
+      movie.description
+        .toLocaleLowerCase()
+        .includes(inputValue.trim().toLocaleLowerCase())
+    );
+  });
+}
```

Comment:
Trim and lowercasing happens on every movie which is not efficient it also happens two times. I would do it like that this looks clearer and more efficient now.

```suggestion
  const trimmedLowercaseInput = inputValue.trim().toLocaleLowerCase()

  return movies.filter(movie => {
    const isTitleMatching =  movie.title.toLocaleLowerCase().includes(trimmedLowercaseInput)
    const isDescriptionMatching =   movie.description.toLocaleLowerCase().includes(trimmedLowercaseInput)

    return isTitleMatching ||  isDescriptionMatching;
  });
}
```